src/server/obj-info.c

```diff
@@ -67,8 +67,69 @@ static cptr act_description[ACT_MAX] =
 	"berserk rage (50+d50 turns)"
 };
 
+/* MAngband specific. Math sourced from cmd6.c. 
+ * Give the player an idea of their odds of successfully activating 
+ * a wand, staff, rod, or item. - Avenger */
+void describe_activation_chance(const object_type *o_ptr)
+{
+	int	chance, lev, maskIndex;
+	char *mask[10] = {
+		"almost no",
+		"a negligible",
+		"a very low",
+		"a low",
+		"a moderate",
+		"a good",
+		"a very good",
+		"an excellent",
+		"an almost certain",
+	};
+	player_type *p_ptr = Players[player_textout];
+	u32b f1, f2, f3;
+
+	/* Extract the flags */
+	object_flags(o_ptr, &f1, &f2, &f3);
+
+	/* Verify that this is a magic device or other item than can be activated */
+	if (o_ptr->tval == TV_STAFF || o_ptr->tval == TV_WAND || o_ptr->tval == TV_ROD || (f3 & TR3_ACTIVATE))
+	{
+		/* Base chance of success */
+		chance = p_ptr->skill_dev;
+
+		/* Extract the item level */
+		lev = k_info[o_ptr->k_idx].level;
 
+		/* Confusion hurts skill */
+		if (p_ptr->confused) chance = chance / 2;
 
+		/* High level objects are harder */
+		chance = chance - ((lev > 50) ? 50 : lev);
+
+		/* Give everyone a (slight) chance */
+		if (chance < USE_DEVICE)
+		{
+			chance = USE_DEVICE;
+		}
+
+		/* Set the descriptive mask.
+		 *
+		 * Because activation failure has a constant minimum threshhold regardless of the value
+		 * of chance, this is not a strict percentage. Instead, I use a Fibonacci sequence to 
+		 * sort the masks into approximate tiers. */
+		if (chance <= USE_DEVICE) maskIndex = 0;
+		else if (chance < 5) maskIndex = 1;
```

Comment:
Interesting use of Fibonacci series.

Reply:
It seems appropriate for the way the percentages scale based on chance - I've calculated the actual odds and made the messages more accurate in the latest changes.